### ISSUE TITLE
error installing scmer

### ISSUE THREAD

**Issue by Kur1sutaru** [KChen-lab/marker-selection#3]
I tried in several machines install scmer but I get this error:

```
ERROR: Failed cleaning build dir for numpy
      Failed to build numpy
      ERROR: Could not build wheels for numpy, which is required to install pyproject.toml-based projects
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

I want to know which requirements we need, in the requirements file we have only the scikit-learn version, could you please provide the correct versions for torch, numpy, scipy, pandas, and seaborn?

Thanks a lot! 😄

**Comment by Kur1sutaru** [comment]
tks

**Comment by lshh125** [comment]
Hi. I guess you might have figured it out. Here are my latest working system & package info in case it is helpful to you or others.

Please don't hesitate to contact me if you have any other questions.

System: Ubuntu 20.04.5 LTS (5.15.0-73-generic)
GPU: RTX 3080 (NVIDIA-SMI 515.105.01   Driver Version: 515.105.01   CUDA Version: 11.7 )

Conda environment:
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                  2_kmp_llvm    conda-forge
anndata                   0.8.0              pyhd8ed1ab_1    conda-forge
anyio                     3.5.0            py39h06a4308_0  
argon2-cffi               20.1.0           py39h27cfd23_1  
arpack                    3.7.0                hc6cf775_2    conda-forge
asttokens                 2.0.5              pyhd3eb1b0_0  
attrs                     21.4.0             pyhd3eb1b0_0  
babel                     2.9.1              pyhd3eb1b0_0  
backcall                  0.2.0              pyhd3eb1b0_0  
beautifulsoup4            4.11.1           py39h06a4308_0  
blas                      1.0                         mkl  
bleach                    4.1.0              pyhd3eb1b0_0  
bottleneck                1.3.5            py39h7deecbd_0  
brotli                    1.0.9                h5eee18b_7  
brotli-bin                1.0.9                h5eee18b_7  
brotlipy                  0.7.0           py39hb9d737c_1004    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
ca-certificates           2023.5.7             hbcca054_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
certifi                   2023.5.7           pyhd8ed1ab_0    conda-forge
cffi                      1.15.1           py39h74dc2b5_0  
charset-normalizer        2.1.0              pyhd8ed1ab_0    conda-forge
cryptography              37.0.4           py39hd97740a_0    conda-forge
cudatoolkit               11.6.0              hecad31d_10    conda-forge
cycler                    0.11.0             pyhd3eb1b0_0  
dbus                      1.13.18              hb2f20db_0  
debugpy                   1.5.1            py39h295c915_0  
decorator                 5.1.1              pyhd3eb1b0_0  
defusedxml                0.7.1              pyhd3eb1b0_0  
dgl                       0.9.1                    py39_0    dglteam
einops                    0.6.1              pyhd8ed1ab_0    conda-forge
entrypoints               0.4              py39h06a4308_0  
executing                 0.8.3              pyhd3eb1b0_0  
expat                     2.4.4                h295c915_0  
ffmpeg                    4.3                  hf484d3e_0    pytorch
filelock                  3.9.0            py39h06a4308_0  
fontconfig                2.13.1               h6c09931_0  
fonttools                 4.25.0             pyhd3eb1b0_0  
freetype                  2.12.1               hca18f0e_0    conda-forge
glib                      2.69.1               h4ff587b_1  
glpk                      5.0                  h445213a_0    conda-forge
gmp                       6.2.1                h58526e2_0    conda-forge
gnutls                    3.6.13               h85f3911_1    conda-forge
gst-plugins-base          1.14.0               hbbd80ab_1  
gstreamer                 1.14.0               h28cd5cc_2  
h5py                      3.7.0            py39h737f45e_0  
hdf5                      1.10.6          nompi_h3c11f04_101    conda-forge
huggingface_hub           0.15.1                     py_0    huggingface
icu                       58.2                 he6710b0_3  
idna                      3.3                pyhd8ed1ab_0    conda-forge
igraph                    0.10.2               ha1f048c_0    conda-forge
importlib-metadata        6.0.0              pyha770c72_0    conda-forge
importlib_metadata        6.0.0                hd8ed1ab_0    conda-forge
ipykernel                 6.9.1            py39h06a4308_0  
ipython                   8.4.0            py39h06a4308_0  
ipython_genutils          0.2.0              pyhd3eb1b0_1  
ipywidgets                7.6.5              pyhd3eb1b0_1  
jedi                      0.18.1           py39h06a4308_1  
jinja2                    3.0.3              pyhd3eb1b0_0  
joblib                    1.1.0              pyhd3eb1b0_0  
jpeg                      9e                   h166bdaf_2    conda-forge
json5                     0.9.6              pyhd3eb1b0_0  
jsonschema                4.4.0            py39h06a4308_0  
jupyter                   1.0.0            py39h06a4308_8  
jupyter_client            7.2.2            py39h06a4308_0  
jupyter_console           6.4.3              pyhd3eb1b0_0  
jupyter_core              4.10.0           py39h06a4308_0  
jupyter_server            1.18.1           py39h06a4308_0  
jupyterlab                3.4.4            py39h06a4308_0  
jupyterlab_pygments       0.1.2                      py_0  
jupyterlab_server         2.12.0           py39h06a4308_0  
jupyterlab_widgets        1.0.0              pyhd3eb1b0_1  
kiwisolver                1.4.2            py39h295c915_0  
lame                      3.100             h7f98852_1001    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
leidenalg                 0.9.1            py39h227be39_0    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
libblas                   3.9.0            12_linux64_mkl    conda-forge
libbrotlicommon           1.0.9                h5eee18b_7  
libbrotlidec              1.0.9                h5eee18b_7  
libbrotlienc              1.0.9                h5eee18b_7  
libcblas                  3.9.0            12_linux64_mkl    conda-forge
libdeflate                1.13                 h166bdaf_0    conda-forge
libffi                    3.3                  he6710b0_2  
libgcc-ng                 12.1.0              h8d9b700_16    conda-forge
libgfortran-ng            7.5.0               ha8ba4b0_17  
libgfortran4              7.5.0               ha8ba4b0_17  
libiconv                  1.17                 h166bdaf_0    conda-forge
liblapack                 3.9.0            12_linux64_mkl    conda-forge
libllvm11                 11.1.0               he0ac6c6_4    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libpng                    1.6.37               h753d276_4    conda-forge
libsodium                 1.0.18               h7b6447c_0  
libsqlite                 3.39.2               h753d276_1    conda-forge
libstdcxx-ng              12.1.0              ha89aaad_16    conda-forge
libtiff                   4.4.0                h0e0dad5_3    conda-forge
libuuid                   1.0.3                h7f8727e_2  
libwebp-base              1.2.4                h166bdaf_0    conda-forge
libxcb                    1.13              h7f98852_1004    conda-forge
libxml2                   2.9.14               h74e7548_0  
libzlib                   1.2.12               h166bdaf_2    conda-forge
llvm-openmp               14.0.4               he0ac6c6_0    conda-forge
llvmlite                  0.39.1           py39h7d9a04d_0    conda-forge
local-attention           1.8.6                    pypi_0    pypi
markupsafe                2.1.1            py39h7f8727e_0  
matplotlib                3.5.1            py39h06a4308_1  
matplotlib-base           3.5.1            py39ha18d171_1  
matplotlib-inline         0.1.2              pyhd3eb1b0_2  
metis                     5.1.0             h58526e2_1006    conda-forge
mistune                   0.8.4           py39h27cfd23_1000  
mkl                       2021.4.0           h8d4b97c_729    conda-forge
mkl-service               2.4.0            py39h7e14d7c_0    conda-forge
mkl_fft                   1.3.1            py39h0c7bc48_1    conda-forge
mkl_random                1.2.2            py39hde0f152_0    conda-forge
mpfr                      4.1.0                h9202a9a_1    conda-forge
munkres                   1.1.4                      py_0  
natsort                   8.2.0              pyhd8ed1ab_0    conda-forge
nbclassic                 0.3.5              pyhd3eb1b0_0  
nbclient                  0.5.13           py39h06a4308_0  
nbconvert                 6.4.4            py39h06a4308_0  
nbformat                  5.3.0            py39h06a4308_0  
ncurses                   6.3                  h27087fc_1    conda-forge
nest-asyncio              1.5.5            py39h06a4308_0  
nettle                    3.6                  he412f7d_0    conda-forge
networkx                  2.8.4            py39h06a4308_0  
notebook                  6.4.12           py39h06a4308_0  
numba                     0.56.4           py39h61ddf18_0    conda-forge
numexpr                   2.8.3            py39h807cd23_0  
numpy                     1.21.5           py39h6c91a56_3  
numpy-base                1.21.5           py39ha15fc14_3  
openh264                  2.1.1                h780b84a_0    conda-forge
openjpeg                  2.5.0                h7d73246_1    conda-forge
openssl                   1.1.1u               hd590300_0    conda-forge
packaging                 21.3               pyhd3eb1b0_0  
pandas                    1.4.3            py39h6a678d5_0  
pandocfilters             1.5.0              pyhd3eb1b0_0  
parso                     0.8.3              pyhd3eb1b0_0  
patsy                     0.5.3              pyhd8ed1ab_0    conda-forge
pcre                      8.45                 h295c915_0  
pexpect                   4.8.0              pyhd3eb1b0_3  
pickleshare               0.7.5           pyhd3eb1b0_1003  
pillow                    9.2.0            py39hd5dbb17_2    conda-forge
pip                       22.2.2             pyhd8ed1ab_0    conda-forge
plotly                    5.9.0            py39h06a4308_0  
prometheus_client         0.14.1           py39h06a4308_0  
prompt-toolkit            3.0.20             pyhd3eb1b0_0  
prompt_toolkit            3.0.20               hd3eb1b0_0  
psutil                    5.9.0            py39h5eee18b_0  
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
ptyprocess                0.7.0              pyhd3eb1b0_2  
pure_eval                 0.2.2              pyhd3eb1b0_0  
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pygments                  2.11.2             pyhd3eb1b0_0  
pynndescent               0.5.8              pyh1a96a4e_0    conda-forge
pyopenssl                 22.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.4              pyhd3eb1b0_0  
pyqt                      5.9.2            py39h2531618_6  
pyrsistent                0.18.0           py39heee7806_0  
pysocks                   1.7.1            py39hf3d152e_5    conda-forge
python                    3.9.12               h12debd9_1  
python-dateutil           2.8.2              pyhd3eb1b0_0  
python-fastjsonschema     2.15.1             pyhd3eb1b0_0  
python-igraph             0.10.1           py39h000617a_1    conda-forge
python_abi                3.9                      2_cp39    conda-forge
pytorch                   1.12.1          py3.9_cuda11.6_cudnn8.3.2_0    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pytz                      2022.1           py39h06a4308_0  
pyyaml                    6.0              py39h5eee18b_1  
pyzmq                     23.2.0           py39h6a678d5_0  
qt                        5.9.7                h5867ecd_1  
qtconsole                 5.3.1            py39h06a4308_1  
qtpy                      2.0.1              pyhd3eb1b0_0  
readline                  8.1.2                h0f457ee_0    conda-forge
regex                     2022.7.9         py39h5eee18b_0  
requests                  2.28.1             pyhd8ed1ab_0    conda-forge
scanpy                    1.9.1              pyhd8ed1ab_0    conda-forge
scikit-learn              0.23.2                   pypi_0    pypi
scipy                     1.7.3            py39hc147768_0  
scmer                     0.1.0a4                  pypi_0    pypi
seaborn                   0.12.1           py39h06a4308_0  
send2trash                1.8.0              pyhd3eb1b0_1  
session-info              1.0.0              pyhd8ed1ab_0    conda-forge
setuptools                65.1.0           py39hf3d152e_0    conda-forge
sip                       4.19.13          py39h295c915_0  
six                       1.16.0             pyh6c4a22f_0    conda-forge
sniffio                   1.2.0            py39h06a4308_1  
soupsieve                 2.3.1              pyhd3eb1b0_0  
sqlite                    3.39.2               h4ff8645_1    conda-forge
stack_data                0.2.0              pyhd3eb1b0_0  
statsmodels               0.13.5           py39h2ae25f5_2    conda-forge
stdlib-list               0.8.0              pyhd8ed1ab_0    conda-forge
suitesparse               5.10.1               h9e50725_1    conda-forge
tbb                       2021.5.0             h924138e_1    conda-forge
tenacity                  8.0.1            py39h06a4308_1  
terminado                 0.13.1           py39h06a4308_0  
testpath                  0.6.0            py39h06a4308_0  
texttable                 1.6.7              pyhd8ed1ab_0    conda-forge
threadpoolctl             2.2.0              pyh0d69192_0  
tk                        8.6.12               h27826a3_0    conda-forge
tokenizers                0.13.2                   py39_0    huggingface
torchaudio                0.12.1               py39_cu116    pytorch
torchvision               0.13.1               py39_cu116    pytorch
tornado                   6.1              py39h27cfd23_0  
tqdm                      4.64.0           py39h06a4308_0  
traitlets                 5.1.1              pyhd3eb1b0_0  
transformers              4.24.0           py39h06a4308_0  
typing-extensions         4.5.0            py39h06a4308_0  
typing_extensions         4.5.0            py39h06a4308_0  
tzdata                    2022c                h191b570_0    conda-forge
umap-learn                0.5.3            py39hf3d152e_0    conda-forge
urllib3                   1.26.11            pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.5              pyhd3eb1b0_0  
webencodings              0.5.1            py39h06a4308_1  
websocket-client          0.58.0           py39h06a4308_4  
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
widgetsnbextension        3.5.2            py39h06a4308_0  
xorg-libxau               1.0.9                h7f98852_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml                      0.2.5                h7b6447c_0  
zeromq                    4.3.4                h2531618_0  
zipp                      3.13.0             pyhd8ed1ab_0    conda-forge
zlib                      1.2.12               h166bdaf_2    conda-forge
zstd                      1.5.2                h6239696_4    conda-forge
